src/graph/edge.cc

```diff
@@ -526,6 +544,11 @@ namespace hpp {
 
       void WaypointEdge::nbWaypoints (const size_type number)
       {
+        if (edges_.size() > 1) {
+          HPP_THROW(std::runtime_error,
+                    "It is not allowed to resize a waypoint edge more than "
+                    "once: nb edges:" << edges_.size());
+        }
```

Comment:
Why is this not allowed ? Calling invalidate should be sufficient, no ?

Reply:
I do not remember why I added this test. You are probably right, calling invalidate should be sufficient.